activerecord/test/cases/associations/has_many_through_associations_test.rb

```diff
@@ -586,6 +586,31 @@ def test_replace_association
     assert_not_includes posts(:welcome).reload.people.reload, people(:michael)
   end
 
+  def test_replace_association_with_duplicates
+    post   = posts(:thinking)
+    person = people(:david)
+
+    assert_difference "post.people.count", 2 do
+      post.people = [person]
+      post.people = [person, person]
+    end
+  end
+
+  def test_intersection_and_difference
+    reflection = ActiveRecord::Reflection.create(:has_many, :clients, nil, {}, Post)
+    association = ActiveRecord::Associations::HasManyThroughAssociation.new(nil, reflection)
+
+    assert_equal association.send(:intersection,
+       [1, 1, 2, 3, 3],
+       [1, 3, 3, 3, 4]
+    ), [1, 3, 3]
+
+    assert_equal association.send(:difference,
+       [1, 1, 2, 3, 3],
+       [1, 3, 3, 3, 4]
+    ), [1, 2]
```

Comment:
I'd prefer we don't use `send` on an internal API.  Is the above integration test not enough to prevent regression?  If not, can we add another integration test?  That way we are free to change the internal API later.

Reply:
Thanks for your review!

Yes, good point. I looked at the code more, and I think we're good with only the integration test.

I tossed the respective commit and pushed again.